zigate/core.py

```diff
@@ -1254,19 +1255,19 @@ def _neighbours_table(self, addr, nodes=None):
         while index < entries:
             r = self.lqi_request(addr, index, True)
             if not r:
-                LOGGER.error('Failed to build neighbours table')
-                return
+                LOGGER.error('Failed to build neighbours table: addr=%s, index=%s', addr, index)
+                return []
             data = r.cleaned_data()
             entries = data['entries']
             for n in data['neighbours']:
                 # bit_field
-                # bit 0-1 = u2RxOnWhenIdle 0/1
-                # bit 2-3 = u2Relationship 0/1/2
-                # bit 4-5 = u2PermitJoining 0/1
-                # bit 6-7 = u2DeviceType 0/1/2
-                is_parent = n['bit_field'][2:4] == '00'
-                is_child = n['bit_field'][2:4] == '01'
-                is_router = n['bit_field'][6:8] == '01'
+                # bit 0-1 = Device Type (0-Coordinator 1-Router 2-End Device)
+                # bit 2-3 = Permit Join status (1- On 0-Off)
+                # bit 4-5 = Relationship (0-Parent 1-Child 2-Sibling)
+                # bit 6-7 = Rx On When Idle status (1-On 0-Off)
+                is_parent = n['bit_field'][4:6] == '00'
```

Comment:
Maybe all those properties could be exposed to the web view to provide more details in the rendering, like specific color and/or shapes.